### PR TITLE
Implemented settings tab for Quiz

### PR DESCRIPTION
**Pull request by abalanovsky**
Implemented Settings Tab for Quiz Settings

https://github.com/ita-social-projects/SpaceToStudy-Client/assets/108689551/5a220007-0686-4c3f-bdb9-d9c70950f469

#1220

### CODE REVIEW

File: src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx [hunk 1]
```diff
@@ -1,7 +1,105 @@
...
+              {t('myResourcesPage.quizzes.pointValuesDesc')}
+            </Typography>
+          </Box>
+          <Switch
+            checked={data.pointValues}
+            data-testid='pointValues-switch'
+            onChange={handleInputChange('pointValues')}
+            sx={defaultStyles.switch}
+          />
+        </Box>
```
Comment (TSlashDreamy) [inline src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx:43]: How do you think, maybe we can make reusable component out of this to not repeat the same pieces of code?

Comment (TSlashDreamy) [reply src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx:43]: > I think .map it's an overkill decision for rendering two different titles and four switches at all.

Yeah, you are right, but I'm talking just about using this newly crafted item, without a `.map` function. Something like this:

```js
<Box>
    <SettingItem propsHere/>
    <SettingItem propsHere/>
    <SettingItem propsHere/>
</Box>
```

What do you think?

File: src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.styles.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { TypographyVariantEnum } from '~/types'
+export const styles = {
+  topTitle: {
+    mt: 1
```
Comment (Tolik170) [inline src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.styles.ts:4]: here you set "1" for mt and below you set "40px", better to use one of these options everywhere. We prefer to use px

File: src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx [hunk 3]
```diff
@@ -1,7 +1,69 @@
...
+      scoredUnscoredResponses: false,
+      correctAnswers: false,
+      shuffleQuestions: false
+    }
+  })
+
+  return (
+    <Box>
+      <Box>
+        <Typography sx={{ ...styles.title, ...styles.topTitle }}>
```
Comment (Tolik170) [inline src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx:23]: Use our spliceSx function to combine both styles:

`<Typography sx={spliceSx(styles.title, styles.topTitle)}`

File: src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx [hunk 4]
```diff
@@ -1,7 +1,69 @@
+import { useTranslation } from 'react-i18next'
 import { Box } from '@mui/material'
+import Typography from '@mui/material/Typography'
+
+import useForm from '~/hooks/use-form'
+import { styles } from '~/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.styles'
```
Comment (Tolik170) [inline src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx:6]: styles should be after SettingItem and divided by space

File: src/components/setting-item/SettingItem.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    event: React.ChangeEvent<HTMLInputElement>,
+    checked: boolean
+  ) => void
+}
+
+const SettingItem: React.FC<SettingItemProps> = ({
+  title,
+  subtitle,
+  checked,
+  dataTestId,
```
Comment (Tolik170) [inline src/components/setting-item/SettingItem.tsx:24]: maybe you should set a default value for dataTestId prop

File: src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx (hunk 1 above)
Comment (abalanovsky) [reply src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx]: U right, thank you, I made changes

**Review by MariaAfonina — APPROVED** [review]

File: src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.styles.ts (hunk 2 above)
Comment (abalanovsky) [reply src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.styles.ts]: Got it

File: src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx (hunk 3 above)
Comment (abalanovsky) [reply src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx]: Thank you, I will use it

File: src/components/setting-item/SettingItem.tsx (hunk 5 above)
Comment (abalanovsky) [reply src/components/setting-item/SettingItem.tsx:24]: I made it optional

**Review by Tolik170 — APPROVED** [review]

**Review by OlyaKorchan — APPROVED** [review]

**Review by MariaAfonina — APPROVED** [review]

**Review by dmtrth25 — APPROVED** [review]

**Review by TSlashDreamy — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Comment by abalanovsky** [comment]
## Changes in issue

https://github.com/ita-social-projects/SpaceToStudy-Client/issues/1220

**Review by Marichka0406 — APPROVED** [review]

**Pull request merged**